pyccl/halos/massdef.py

```diff
@@ -265,7 +272,43 @@ class MassDefVir(MassDef):
     Args:
         c_m (string): concentration-mass relation.
     """
+    name = 'vir'
+
     def __init__(self, c_m='Klypin11'):
         super(MassDefVir, self).__init__('vir',
                                          'critical',
                                          c_m_relation=c_m)
+
+
+def mass_def_from_name(name):
+    """ Return mass definition subclass from name string.
+
+    Args:
+        name (string): a mass function name
+
+    Returns:
+        MassDef subclass corresponding to the input name.
+    """
+    if name == "vir":
+        c_m_relation = "Klypin11"
+        return MassDefVir
```

Comment:
you don't seem to be using `c_m_relation` for anything

Reply:
OK